modules/nw-autoscaling-ingress-controller.adoc

```diff
@@ -0,0 +1,222 @@
+      key: ca.crt
+EOF
+----
+
+. Create a role for reading metrics from Thanos by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f - <<EOF
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: thanos-metrics-reader
+rules:
+- apiGroups:
+  - ""
+  resources:
+  - pods
+  - nodes
+  verbs:
+  - get
+- apiGroups:
+  - metrics.k8s.io
+  resources:
+  - pods
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - ""
+  resources:
+  - namespaces
+  verbs:
+  - get
+EOF
+----
+
+. Add the new role to the Service Account by entering the following commands:
++
+[source,terminal]
+----
+$ oc adm policy add-role-to-user thanos-metrics-reader -z thanos --role=namespace=openshift-ingress-operator
+----
++
+[source,terminal]
+----
+$ oc adm policy -n openshift-ingress-operator add-cluster-role-to-user cluster-monitoring-view -z thanos
+----
++
+[NOTE]
+====
+The argument `add-cluster-role-to-user` is only required if you use cross-namespace queries. The following step uses a query from the `kube-metrics` namespace which requires this argument.
+====
+
+. Create a new `ScaledObject` YAML file, `ingress-autoscaler.yaml`, that targets the default Ingress Controller deployment:
++
+.Example `ScaledObject` definition
+[source,yaml]
+----
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: ingress-scaler
+spec:
+  scaleTargetRef: <1>
+    apiVersion: operator.openshift.io/v1
+    kind: IngressController
+    name: default
+    envSourceContainerName: ingress-operator
+  minReplicaCount: 1
+  maxReplicaCount: 20 <2>
+  cooldownPeriod: 1
+  pollingInterval: 1
+  triggers:
+  - type: prometheus
+    metricType: AverageValue
+    metadata:
+      serverAddress: https://<example-cluster>:9091 <3>
+      namespace: openshift-ingress-operator <4>
+      metricName: 'kube-node-role'
+      threshold: '1'
+      query: 'sum(kube_node_role{role="worker",service="kube-state-metrics"})' <5>
+      authModes: "bearer"
+    authenticationRef:
+      name: keda-trigger-auth-prometheus
+----
+<1> The custom resource that you are targeting. In this case, the Ingress Controller.
+<2> Optional: the maximum number of replicas. If you omit this field, the default maximum is set to 50 replicas.
+<3> The cluster address and port.
+<4> The Ingress Operator namespace.
+<5> This expression evaluates to however many worker nodes are present in the deployed cluster.
++
+[IMPORTANT]
+====
+If you are using cross-namespace queries you must target port 9091 and not port 9092 in the `serverAddress` field. You also must have elevated privileges to read metrics from this port.
+====
+
+. Apply the custom resource defintion by running the following command:
++
+[source,terminal]
+----
+$ oc apply -f ingress-autoscaler.yaml
+----
+
+.Verification
+* Verify that the default Ingress Controller is scaled out to match the value returned by the `kube-state-metrics` query by running the following commands:
```

Comment:
I wonder if there is a way you can break up this verification section since there are two commands  and two outputs here. As it renders it almost looks like the second verification command is part of the first example output. Maybe a. and b. or bullets?

Reply:
Yeah I see what you mean. I tried with some bullets, lmk what you think